#
# This patch contains changes for Solaris build.   This is for Solaris only,
# so we will not contribute back this change to the upstream community.
#
--- old/src/pam_pkcs11/pam_pkcs11.c	2022-08-04 14:36:46.725084821 +0200
+++ new/src/pam_pkcs11/pam_pkcs11.c	2022-08-04 14:46:04.216029288 +0200
@@ -112,9 +112,7 @@ static int pam_prompt(pam_handle_t *pamh
   free(&resp[0]);
   return PAM_SUCCESS;
 }
-#endif
 
-#if !defined(HAVE_SECURITY_PAM_EXT_H) || defined(OPENPAM)
 static void
 pam_syslog(pam_handle_t *pamh, int priority, const char *fmt, ...)
 {
@@ -124,7 +122,9 @@ pam_syslog(pam_handle_t *pamh, int prior
     vsyslog(priority, fmt, ap);
     va_end(ap);
 }
+#endif
 
+#if !defined(HAVE_SECURITY_PAM_EXT_H) && defined(OPENPAM)
 /*
  * With OpenPAM pam_prompt resp arg cannot be NULL, so this is just a wrapper.
  */
